### ISSUE TITLE
File Read Error

### ISSUE THREAD

**Issue by schlegelp** [seung-lab/cloud-volume#293]
Hi there! We're using this brilliant tool to access publicly hosted segmentation data. 

The following code works perfectly fine for me...
```Python
from cloudvolume import CloudVolume
# Initialise cloudvolume
vol = CloudVolume("https://storage.googleapis.com/fafb-ffn1-20190805/segmentation")
# Try accessing a single location
vol[[100000, 30000, 3000]]
```
... but @pgibb96 encounters a `File Read Error`:
```Python
File Read Error: 99964 bytes, Bbox([99968, 29952, 2976],[100096, 30080, 3008], dtype=int32), Bbox([99968, 29952, 2976],[100096, 30080, 3008], dtype=int32), errors: could not broadcast input array from shape (2,16,2) into shape (2,16,16)
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-18-1e02ec0d5e7c> in <module>
      3 vol = CloudVolume("https://storage.googleapis.com/fafb-ffn1-20190805/segmentation")
      4 # Try accessing a single location
----> 5 vol[[100000, 30000, 3000]]
/usr/local/lib/python3.7/site-packages/cloudvolume/frontends/precomputed.py in __getitem__(self, slices)
    519     requested_bbox = Bbox.from_slices(slices)
    520 
--> 521     img = self.download(requested_bbox, self.mip)
    522     return img[::steps.x, ::steps.y, ::steps.z, channel_slice]
    523 
/usr/local/lib/python3.7/site-packages/cloudvolume/frontends/precomputed.py in download(self, bbox, mip, parallel, segids, preserve_zeros)
    555       parallel = self.parallel
    556 
--> 557     img = self.image.download(bbox, mip, parallel=parallel)
    558 
    559     if segids is None:
/usr/local/lib/python3.7/site-packages/cloudvolume/datasource/precomputed/image/__init__.py in download(self, bbox, mip, parallel, location, retain, use_shared_memory, use_file, order)
    120         progress=self.config.progress,
    121         fill_missing=self.fill_missing,
--> 122         order=order,
    123       )
    124     else:
/usr/local/lib/python3.7/site-packages/cloudvolume/datasource/precomputed/image/rx.py in download_sharded(requested_bbox, mip, meta, cache, spec, compress, progress, fill_missing, order)
     72     img3d = decode(
     73       meta, cutout_bbox,
---> 74       chunkdata, fill_missing, mip
     75     )
     76 
/usr/local/lib/python3.7/site-packages/cloudvolume/datasource/precomputed/image/rx.py in decode(meta, input_bbox, content, fill_missing, mip)
    320       shape=shape,
    321       dtype=meta.dtype,
--> 322       block_size=meta.compressed_segmentation_block_size(mip),
    323     )
    324   except Exception as error:
/usr/local/lib/python3.7/site-packages/cloudvolume/chunks.py in decode(filedata, encoding, shape, dtype, block_size)
     74     return fpzip.decompress(filedata, order='F')
     75   elif encoding == "compressed_segmentation":
---> 76     return decode_compressed_segmentation(filedata, shape=shape, dtype=dtype, block_size=block_size)
     77   elif encoding == "jpeg":
     78     return decode_jpeg(filedata, shape=shape, dtype=dtype)
/usr/local/lib/python3.7/site-packages/cloudvolume/chunks.py in decode_compressed_segmentation(bytestring, shape, dtype, block_size, accelerated)
    165     return decode_compressed_segmentation_c_ext(bytestring, shape, dtype, block_size)
    166 
--> 167   return decode_compressed_segmentation_pure_python(bytestring, shape, dtype, block_size)
    168 
    169 def decode_compressed_segmentation_c_ext(bytestring, shape, dtype, block_size):
/usr/local/lib/python3.7/site-packages/cloudvolume/chunks.py in decode_compressed_segmentation_pure_python(bytestring, shape, dtype, block_size)
    172 def decode_compressed_segmentation_pure_python(bytestring, shape, dtype, block_size):
    173   chunk = np.empty(shape=shape[::-1], dtype=dtype)
--> 174   csegpy.decode_chunk_into(chunk, bytestring, block_size=block_size)
    175   return chunk.T
    176 
/usr/local/lib/python3.7/site-packages/cloudvolume/py_compressed_segmentation.py in decode_chunk_into(chunk, buf, block_size)
    155                                      "is too large (truncated file?)")
    156         _decode_channel_into(
--> 157             chunk, channel, buf[offset:next_offset], block_size
    158         )
    159 
/usr/local/lib/python3.7/site-packages/cloudvolume/py_compressed_segmentation.py in _decode_channel_into(chunk, channel, buf, block_size)
    224             y*block_size[1] : (y+1)*block_size[1],
    225             x*block_size[0] : (x+1)*block_size[0]
--> 226         ] = block[:zmax, :ymax, :xmax]
    227 
    228 
ValueError: could not broadcast input array from shape (2,16,2) into shape (2,16,16)
```

We're both running cloud-volume `0.65.5`. Probably irrelevant but the only difference I am aware of is that I'm running Python `3.6.7` and he is running `3.7.3`. 

Any help with debugging this would be greatly appreciated :)

**Comment by william-silversmith** [comment]
Hi Phillip, 

I'm noticing that you're using the pure python codec for compressed segmentation here. I'm guessing you guys are running MacOS. I checked the C++ compressed_segmentation PyPI page and it looks like I never uploaded a 3.7 binary for Mac. I just uploaded binaries for 3.7 and 3.8. 

https://pypi.org/project/compressed-segmentation/#files

Try running `pip install compessed_segmentation --upgrade` and see if that fixes the problem. The pure python version is less battle tested, and I suppose it broke in this case.

Will
